measuring_ci/artifacts.py

```diff
@@ -0,0 +1,139 @@
+import asyncio
+import logging
+import os
+from functools import partial
+
+import aiohttp
+import boto3
+import dateutil.parser
+from taskcluster.aio import Queue
+
+from taskhuddler.utils import tc_options
+
+from .utils import semaphore_wrapper
+
+log = logging.getLogger(__name__)
+
+
+async def get_tc_run_artifacts(taskid, runid):
+    log.debug('Fetching TC artifact info for %s/%s', taskid, runid)
+    artifacts = []
+    query = {}
+    async with aiohttp.ClientSession() as session:
+        queue = Queue(options=tc_options(), session=session)
+        while True:
+            resp = await queue.listArtifacts(taskid, runid, query=query)
+
+            # Ammend the artifact information with the task and run ids
+            # to make it easy to find the corresponding S3 object
+            for a in resp['artifacts']:
+                a['_name'] = f'{taskid}/{runid}/{a["name"]}'
+                artifacts.append(a)
+            if 'continuationToken' in resp:
+                query.update({'continuationToken': resp['continuationToken']})
+            else:
+                break
+
+    return artifacts
+
+
+async def get_s3_task_artifacts(taskid,
+                                bucket_name='taskcluster-public-artifacts',
+                                s3_client=None):
+
+    if s3_client is None:
+        s3_client = boto3.client(
+            's3',
+            aws_access_key_id=os.environ.get('TASKCLUSTER_S3_ACCESS_KEY'),
+            aws_secret_access_key=os.environ.get('TASKCLUSTER_S3_SECRET_KEY'),
+        )
+    loop = asyncio.get_event_loop()
+    log.debug('Fetching S3 artifact info for %s', taskid)
+    artifacts = []
+
+    cont_token = None
+    prefix = taskid + '/'
+    while True:
```

Comment:
this code is common between this function and utils.py:find_staged_data_files. maybe we can factor out the s3 logic here?